include/operations/extension_trees.h

```diff
@@ -0,0 +1,103 @@
+/***************************************************************************
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename extension_trees.h
+ *
+ **************************************************************************/
+
+#ifndef SYCL_BLAS_EXTENSION_TREES_H
+#define SYCL_BLAS_EXTENSION_TREES_H
+
+#include <CL/sycl.hpp>
+
+namespace blas {
+
+/*!
+ * @brief Determines which type of reduction to perform
+ */
+enum class Reduction_t : int {
+  full = 0,  // Not implemented yet
```

Comment:
Is this PR waiting on other PRs before it can be merged and working?

Reply:
The ones for the full reduction and the reduction of the columns will come later if ever. They're not used in any blas algorithm yet, just a nice-to-have.

Do you think these lines in the enum should be commented out?

Side note: the PR is waiting on #174 though, but #174 is not that big of a change

Reply:
#174 merged. This PR doesn't have any more dependencies.